mlflow/models/signature.py

```diff
@@ -395,8 +395,14 @@ def _infer_signature_from_input_example(
             signature = ModelSignature(inputs=input_schema, params=params_schema)
         else:
             signature = ModelSignature(input_schema, output_schema, params_schema)
-        if return_prediction:
-            return signature, prediction
+
+        # try assign output schema if failing to infer it from prediction
+        if signature.outputs is None:
```

Comment:
Can we first check if the instance is `_LangChainModelWrapper`? (`mlflow.langchain.utils.chat` imports LangChain, so it will fail for other flavors.)

Reply:
thanks for the catch!